src/sparseml/pytorch/image_classification/utils/helpers.py

```diff
@@ -370,13 +370,14 @@ def save_model_training(
     :param arch_key: if provided, the `arch_key` will be saved in the
         checkpoint
     """
-    has_top1 = "top1acc" in val_res.results
-    metric_name = "top-1 accuracy" if has_top1 else "val_loss"
-    metric = val_res.result_mean("top1acc" if has_top1 else DEFAULT_LOSS_KEY).item()
-    print(
-        f"Saving model for epoch {epoch} and {metric_name} "
-        f"{metric} to {save_dir} for {save_name}"
-    )
+    if val_res is not None:
```

Comment:
when does this come up?

Reply:
The val_res are typed as `Optional[ModuleRunResults]` hence the check enforces certain operations to be performed only when `val_res` is `non-None`. 

Noting: That this is a very rare situation. For me the situation arose with a somewhat faulty recipe when a quantization modifier was present without any actual training, i.e
```yaml
quantization_start_epoch: 0

quantization_modifiers:
  - !QuantizationModifier
    start_epoch: eval(quantization_start_epoch)
```

Omitting this check for the specified recipe will throw an `AttributeError: 'NoneType' object has no attribute 'results'` Which I feel is the wrong error message for this situation. With this check in place the program exits elegantly with no training performed. Any thoughts/suggestions are welcome